sqlalchemy_ibmi/base.py

```diff
@@ -772,13 +772,23 @@ def create_connect_args(self, url):
         opts = url.translate_connect_args(username='user', host='system')
         opts.update(url.query)
         allowed_opts = {'system', 'user', 'password',
-                        'autocommit', 'readonly', 'timeout', 'database'}
+                        'autocommit', 'readonly', 'timeout',
+                        'database', 'library_list', 'current_schema'}
         if allowed_opts < opts.keys():
             raise ValueError("Option entered not valid for "
                              "IBM i Access ODBC Driver")
- 
-        return [["Driver={%s}; UNICODESQL=1; TRUEAUTOCOMMIT=1" % (
-                 self.pyodbc_driver_name)], opts]
+
+        connection_string = "Driver={%s}; UNICODESQL=1; TRUEAUTOCOMMIT=1;" % (
+                 self.pyodbc_driver_name)
+        libl = 'DefaultLibraries='
+        if 'current_schema' in opts:
+            libl = libl + opts.current_schema + ','
```

Comment:
Are the options also attributes or is opts just a dict? If it's just a dict, you'd need to do `opts['current_schema']`

Reply:
Also, since we will have a separate current_schema setting it should always be responsible for setting the current schema. Currently, if it is not set, but a library_list _is_ set, the first entry of the library list will be set to the current schema. Instead, I think if the user does not set a current_schema there should be no current schema set, so an empty comma should be placed at the front of DefaultLibraries.

We can do this rather simply with something like
```python
libl = 'DefaultLibraries={},'.format(opts.get('current_schema', ''))